data-prepper-plugins/grok-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/grok/GrokProcessor.java

```diff
@@ -97,6 +98,7 @@ public GrokProcessor(final PluginSetting pluginSetting, final ExpressionEvaluato
         this.fieldToGrok = new LinkedHashMap<>();
         this.executorService = executorService;
         this.expressionEvaluator = expressionEvaluator;
+        this.matchFailTag = grokProcessorConfig.getMatchFailTag();
 
         grokProcessingMatchCounter = pluginMetrics.counter(GROK_PROCESSING_MATCH);
         grokProcessingMismatchCounter = pluginMetrics.counter(GROK_PROCESSING_MISMATCH);
```

Comment:
Should we tag when there is an exception thrown as well (lines 136 - 148), but especially on the last catch block. sometimes exceptions are thrown because of what's in the Event and that is technically a grok match failure.

Reply:
Sure. We can add the same tag. Or a different tag. This is something we should discuss.